test/examples/examples.spec.js

```diff
@@ -0,0 +1,168 @@
+            detached: true,
+            stdio: ['ignore', 'pipe', 'pipe']
+        }, options);
+
+        var spawned = spawn(command, args, options);
+        var stdout = '';
+        var stderr = '';
+
+        spawned.stdout.on('data', (data) => {
+            logger.log('STDOUT:', data);
+            stdout += data;
+        });
+
+        spawned.stderr.on('data', (data) => {
+            logger.log('STDERR:', data);
+            stderr += data;
+        });
+
+        spawned.on('close', (code) => {
+            if (code === 0) {
+                resolve(stdout, stderr);
+            } else {
+                reject(Error('command: "' + command + ' ' + args.join(' ') +
+                             '", failed with ' + code));
+            }
+        });
+    });
+}
+
+describe('examples', () => {
+    let hostAddress = 'localhost';
+    let juttleEnginePort = 8080;
+    let baseUrl = `http://${hostAddress}:${juttleEnginePort}`
+    let baseAPIUrl = `${baseUrl}/api/v0`
+
+    let retryHTTPOptions = {
+        interval: 500,
+        timeout: 10000
+    }
+
+    before(() => {
+        return execAsync('docker run ubuntu route | grep default | awk \'{print $2}\'')
+        .then((stdout) => {
+            hostAddress = stdout.trim();
+            return spawnify('docker', ['build', '-q', '-t', 'juttle/juttle-engine:latest', '.']);
+        })
+        .then(() => {
+            return spawnify('test/examples/start-example-containers.sh');
+        })
+        .then(() => {
+            return retry(() => {
+                return new Promise((resolve, reject) => {
+                    http.get(baseUrl, (resp) => {
+                        if (resp.statusCode !== 200) {
+                            reject(Error(`juttle-engine not responding on ${baseUrl}`));
+                        } else {
+                            resolve();
+                        }
+                    })
+                    .on('error', (err) => {
+                        reject(err);
+                    });
+                });
+            }, retryHTTPOptions);
+        });
+    });
+
+    after(() => {
+        // shutdown and remove containers
+        return spawnify('test/examples/stop-example-containers.sh');
+    });
+
+    _.each({
+        'core-juttle': 'examples/core-juttle',
+        'cadvisor-influx': 'examples/cadvisor-influx/'
+    }, (directory, name) => {
+        let juttles = fs.readdirSync(directory);
+        juttles = juttles.filter((file) => {
+            return file.endsWith('.juttle');
+        });
+
+        _.each(juttles, (filename) => {
+            it(`can run the juttle "${directory}/${filename}" without errors or warnings`, () => {
+                var jobId;
+                var errors = [];
+                var warnings = [];
+                var views = {};
+                var waitForData;
+
+                return chakram.post(baseAPIUrl + '/jobs/', {
+                    path: path.join(directory, filename)
```

Comment:
Did you try to use the immediate form (e.g. wait: true)? That would save some time mucking around with websockets.